hist/hist/src/TH1.cxx

```diff
@@ -6781,36 +6782,27 @@ void  TH1::SmoothArray(Int_t nn, Double_t *xx, Int_t ntimes)
             // do all elements beside the first and last point for median 3
             //  and first two and last 2 for median 5
             for  ( ii = ifirst; ii < ilast; ii++)  {
-               assert(ii - ifirst >= 0);
-               for  (int jj = 0; jj < medianType; jj++)   {
-                  hh[jj] = yy[ii - ifirst + jj ];
-               }
-               zz[ii] = TMath::Median(medianType, hh);
+               zz[ii] = TMath::Median(medianType, yy.data() + ii - ifirst);
             }
 
```

Comment:
Should not be here `if (kk != 1)` ? 
Otherwise in the case `kk = 2`  the first and last point are not used ? 

And do we have a test for this function ? Maybe we should add one

Reply:
As we figured out in the offline chat, the first and last point stay the same in the third iteration. I will add a comment to clarify that.